### PR TITLE
Create activity files extension

### PR DESCRIPTION
**Pull request by nickvergessen**
Step 1 from #225

### CODE REVIEW

**Comment by nickvergessen** [comment]
Should be ready for the first step, @LukasReschke @DeepDiver1975

**Comment by LukasReschke** [comment]
Sharing notifications are broken, when sharing a file with another user the user gets an error shown (file gets still shared though) and it is also not written as activity.

Error log says `Undefined class constant 'TYPE_SHARED' at /Users/lreschke/Programming/core/apps/activity/lib/hooks.php#264`

**Comment by nickvergessen** [comment]
Fixed the error, the missing activities for group shares themselves are fixed by https://github.com/owncloud/core/pull/13979

File: extension/files.php [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+	protected $l;
+
+	/** @var IURLGenerator */
+	protected $URLGenerator;
+
+	/**
+	 * @param IL10N $l
+	 * @param IURLGenerator $URLGenerator
+	 */
+	public function __construct(IL10N $l, $URLGenerator) {
```
Comment (LukasReschke) [inline extension/files.php]: Add IURLGenerator to CTR?

File: extension/files_sharing.php [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	protected $l;
+
+	/** @var IURLGenerator */
+	protected $URLGenerator;
+
+	/**
+	 * @param IL10N $l
+	 * @param IURLGenerator $URLGenerator
+	 */
+	public function __construct(IL10N $l, $URLGenerator) {
```
Comment (LukasReschke) [inline extension/files_sharing.php]: also here you could typehint the urlgenerator

File: lib/datahelper.php [hunk 3]
```diff
@@ -33,10 +34,10 @@ class DataHelper {
 	/** @var \OCA\Activity\ParameterHelper */
 	protected $parameterHelper;
 
-	/** @var \OC_L10N */
+	/** @var IL10N */
 	protected $l;
 
-	public function __construct(IManager $activityManager, ParameterHelper $parameterHelper, \OC_L10N $l) {
+	public function __construct(IManager $activityManager, ParameterHelper $parameterHelper, IL10N $l) {
```
Comment (LukasReschke) [inline lib/datahelper.php]: PHPDoc?

**Comment by LukasReschke** [comment]
Besides my nitpicks: Code looks great and works :+1:

File: extension/files.php (hunk 1 above)
Comment (nickvergessen) [inline extension/files.php]: file is temporarily only anyway

File: extension/files_sharing.php (hunk 2 above)
Comment (nickvergessen) [inline extension/files_sharing.php]: already done on the sharing PR

**Pull request merged**